File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p03723/p03723_num6_parsed.py
a, b, c = map(int, input().split())

count = 0
while True:
    if a % 2 != 0 or b % 2 != 0 or c % 2 != 0:
        break

    half_a = a // 2
    half_b = b // 2
    half_c = c // 2

    a = half_b + half_c
    b = half_a + half_c
    c = half_a + half_b

    count += 1

if a % 2 != 0 or b % 2 != 0 or c % 2 != 0:
    print(count)
else:
    print(-1)

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p03723/p03723_num6_parsed.cpp
#include<iostream>
using namespace std;

int main() {
    int A, B, C;
    cin >> A >> B >> C;

    int cnt = 0;

    while (true) {
        if (A % 2 != 0 || B % 2 != 0 || C % 2 != 0) {
            break;
        }

        int halfA = A / 2;
        int halfB = B / 2;
        int halfC = C / 2;

        A = halfB + halfC;
        B = halfA + halfC;
        C = halfA + halfB;

        cnt++;
        
        if (A == B && B == C) {
            cnt = -1;
            break;
        }
    }

    cout << cnt << endl;

    return 0;
}